src/main/scala/tech/cryptonomic/conseil/config/LorreAppConfig.scala

```diff
@@ -0,0 +1,67 @@
+package tech.cryptonomic.conseil.config
+
+import tech.cryptonomic.conseil.config.Platforms._
+import tech.cryptonomic.conseil.util.ConfigUtil.Pureconfig.loadAkkaStreamingClientConfig
+import pureconfig.{ConfigFieldMapping, CamelCase, loadConfig}
+import pureconfig.generic.auto._
+import pureconfig.generic.ProductHint
+import pureconfig.error.{ConfigReaderFailures, ConfigReaderFailure}
+
+/** wraps all configuration needed to run Lorre */
+trait LorreAppConfig {
+  import LorreAppConfig._
+
+  /** Reads all configuration upstart, will print all errors encountered during loading */
+  protected def loadApplicationConfiguration(commandLineArgs: Array[String]): Either[ConfigReaderFailures, CombinedConfiguration] = {
+    //applies convention to uses CamelCase when reading config fields
+    implicit def hint[T] = ProductHint[T](ConfigFieldMapping(CamelCase, CamelCase))
+
+    val loadedConf = for {
+      network <- readArgs(commandLineArgs)
+      lorre <- loadConfig[LorreConfiguration](namespace = "lorre")
+      nodeRequests <- loadConfig[NetworkCallsConfiguration]("")
+      node <- loadConfig[TezosNodeConfiguration](namespace = s"platforms.tezos.$network.node")
+      streamingClient <- loadAkkaStreamingClientConfig(namespace = "akka.tezos-streaming-client")
+      sodium <- loadConfig[SodiumConfiguration](namespace = "sodium.libraryPath")
```

Comment:
Sodium config is needed for Conseil and not Lorre. libSodium is used to cryptographically sign blockchain transactions. In our case, clients used to be able to make a Conseil call for transaction signing. We took it out in order to avoid security issues with our launch. However, we plan to bring it back for those using the JAR (i.e. running locally) but not for those using the API (client / server comms).

Reply:
I've moved the dependency to a "send-enabled" subclass of the TezosNodeOperator.
This way:
- Lorre doesn't need the dependency
- Conseil will use Sodium only when we introduce the new operations
- the NodeOperator class is splitted in separate concerns (formally, read vs. write), which helps reduce bloating